app/renderer/js/utils/logger-util.js

```diff
@@ -30,6 +30,20 @@ class Logger {
 			timestamp = this.getTimestamp;
 		}
 
+		fs.readFile(file, 'utf8', (err, data) => {
+			if (err) {
+				throw err;
+			}
+			const logs = data.split('\n');
+			// Keep top 300 files of log for each instance
+			const trimLines = logs.slice(Math.max(logs.length - 300, 0));
+			let toWrite = '';
+			trimLines.forEach(line => {
```

Comment:
I am not sure, but is there a better algorithmic way to this? I think this is of high complexity and could slow the initial startup.

Reply:
Have to think about this.. what do u think @priyank-p ?

Reply:
For the slow down issue, I think it would be one time impact since once we run this code the first time the log file shouldn't be huge. But let's extract this feature into its own function and call it after the `load` event in `main.js`. That would take care of this. Or another thing to try out would be `requestIdleCallback()`.

Reply:
If we keep the function to be an asynchronous one, the performance should also be better.

Reply:
@abhigyank that is the exact reason why i used the synchronous one here..

Reply:
@priyank-p if i call it after the load event in ```main.js``` , how do i exactly make it run for all log files. Do i loop over all log files in log directory? Will that be feasible?

Reply:
Ah didn't think that through. First, extract this feature into its own function then in the constructor check if it is the renderer process. If it renderer process use `requestIdleCallback()` otherwise use `process.nextTick` for main process.

(Can you check if `requestIdleCallback` is available in the main process of Electron? If so we should use that.)

Reply:
```requestIdleCallback``` doesn't seem to be working for the main process. Will use process.nextTick.

Reply:
@priyank-p Using ```request.IdleCallback``` for the renderer process hang up with no domains loaded. While using only process.nextTick works perfectly. I am attaching screenshots for reference. I have pushed the code only for using process.nextTick for both renderer and main process. I suppose it can be used for both. I don't know why requestIdleCallback doesn't work. 
![Screenshot from 2019-04-20 04-21-37](https://user-images.githubusercontent.com/21038781/56447143-4474f980-6324-11e9-80af-94b06ae3477a.png)
![Screenshot from 2019-04-20 04-21-15](https://user-images.githubusercontent.com/21038781/56447144-450d9000-6324-11e9-9826-6cb46588e659.png)

Reply:
OK, I see the issue. In the screenshot you attached the code shows `requestIdleCallback(this.trimLog(file))` which is incorrect it should be `requestIdleCallback(() => this.trimLog(file))`. I bet there is error in the console about that you missed. The problem is that you are calling the function in the paramter which return null which is passed in to `requestIdleCallback` which in turn throws error hence the failure of app to load.

Reply:
oh, let me test that.

Reply:
@priyank-p works fine now. Thanks for your help. Tested for both main and renderer log files. Pushed the changes.